<a href="https://colab.research.google.com/github/Juribe7980/Project-1-Part-1/blob/main/Project_1_Part_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [13]:
import pandas as pd
import numpy as np

In [14]:
Sales = pd.read_csv('/content/drive/MyDrive/Coding Dojo/sales_predictions.csv')


1) How many rows and columns?

In [15]:
Sales.shape

(8523, 12)

There are 8523 rows and 12 columns.

2) What are the datatypes of each variable?

In [16]:
Sales.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

Here are the data types of each variable.

3) Are there duplicates? If so, drop any duplicates.

In [17]:
Sales.duplicated().any()

False

There are no duplicates in this dataset.

4) Identify missing values.

In [18]:
Sales.isna().any()

Item_Identifier              False
Item_Weight                   True
Item_Fat_Content             False
Item_Visibility              False
Item_Type                    False
Item_MRP                     False
Outlet_Identifier            False
Outlet_Establishment_Year    False
Outlet_Size                   True
Outlet_Location_Type         False
Outlet_Type                  False
Item_Outlet_Sales            False
dtype: bool

Item_Weight and Outlet_Size have nas present

5) Decide on how to address the missing values and do it! (This requires your judgement, so explain your choice).

In [19]:
Sales['Item_Weight'].fillna(value =  Sales['Item_Weight'].mean(),inplace = True)
Sales['Outlet_Size'].fillna(value = 'Medium', inplace = True)

6) Confirm that there are no missing values after addressing them.

In [20]:
Sales.isna().any()

Item_Identifier              False
Item_Weight                  False
Item_Fat_Content             False
Item_Visibility              False
Item_Type                    False
Item_MRP                     False
Outlet_Identifier            False
Outlet_Establishment_Year    False
Outlet_Size                  False
Outlet_Location_Type         False
Outlet_Type                  False
Item_Outlet_Sales            False
dtype: bool

7) Find and fix any inconsistent categories of data (example: fix cat, Cat, and cats so that they are consistent) 

In [21]:
Sales['Item_Fat_Content'].value_counts()

Low Fat    5089
Regular    2889
LF          316
reg         117
low fat     112
Name: Item_Fat_Content, dtype: int64

In [22]:
Sales['Item_Fat_Content'].replace(['LF', 'low fat'], 'Low Fat', inplace=True)
Sales['Item_Fat_Content'].replace(['reg'], 'Regular', inplace=True)

In [23]:
Sales['Item_Fat_Content'].value_counts()

Low Fat    5517
Regular    3006
Name: Item_Fat_Content, dtype: int64

In [24]:
Sales['Item_Type'].value_counts()

Fruits and Vegetables    1232
Snack Foods              1200
Household                 910
Frozen Foods              856
Dairy                     682
Canned                    649
Baking Goods              648
Health and Hygiene        520
Soft Drinks               445
Meat                      425
Breads                    251
Hard Drinks               214
Others                    169
Starchy Foods             148
Breakfast                 110
Seafood                    64
Name: Item_Type, dtype: int64

In [25]:
Sales['Outlet_Size'].value_counts()

Medium    5203
Small     2388
High       932
Name: Outlet_Size, dtype: int64

In [26]:
Sales['Outlet_Size'].replace(['High'], 'Large', inplace=True)

In [27]:
Sales['Outlet_Size'].value_counts()

Medium    5203
Small     2388
Large      932
Name: Outlet_Size, dtype: int64

In [28]:
Sales['Outlet_Location_Type'].value_counts()

Tier 3    3350
Tier 2    2785
Tier 1    2388
Name: Outlet_Location_Type, dtype: int64

In [29]:
Sales['Outlet_Type'].value_counts()

Supermarket Type1    5577
Grocery Store        1083
Supermarket Type3     935
Supermarket Type2     928
Name: Outlet_Type, dtype: int64

8) For any numerical columns, obtain the summary statistics of each (min, max, mean)

In [30]:
print(Sales['Item_Weight'].min())
print(Sales['Item_Visibility'].min())
print(Sales['Item_MRP'].min())
print(Sales['Item_Outlet_Sales'].min())
print(Sales['Outlet_Establishment_Year'].min())

4.555
0.0
31.29
33.29
1985


In [31]:
print(Sales['Item_Weight'].max())
print(Sales['Item_Visibility'].max())
print(Sales['Item_MRP'].max())
print(Sales['Item_Outlet_Sales'].max())
print(Sales['Outlet_Establishment_Year'].max())

21.35
0.328390948
266.8884
13086.9648
2009


In [32]:
print(Sales['Item_Weight'].mean())
print(Sales['Item_Visibility'].mean())
print(Sales['Item_MRP'].mean())
print(Sales['Item_Outlet_Sales'].mean())
print(Sales['Outlet_Establishment_Year'].mean())

12.857645184135976
0.06613202877895108
140.9927819781767
2181.288913575032
1997.8318667135984
